### PR TITLE
Porting changes and fixes to azure-support branch

### PR DESCRIPTION
**Pull request by SauarvD**
### Requirements

* Including syncing of files using azcopy
* Bug fixes
* Upgrading to latest sdk

### Description of the Change

* Changes to website build pack for syncing the blobs. Added a separate function for the same.
* Changes to build-module for bug fixes
* Changes to jazz_azure-create-service for upgrading to latest SDK

### CODE REVIEW

File: builds/service-onboarding-build-pack/Jenkinsfile [hunk 1]
```diff
@@ -332,175 +259,156 @@ node  {
...
+			[credentialsId: repo_credential_id, url: buildModuleUrl]
+		]])
 
 		configModule = load "config-loader.groovy"
 		configLoader = configModule.loadConfigData(aws_credential_id, region, instance_prefix)
 		echo "config loader loaded successfully."
 
 		awsAPIGatewayModule = load "aws-apigateway-module.groovy"
-    	awsAPIGatewayModule.initialize()
+    awsAPIGatewayModule.initialize()
```
Comment (devsatishm) [inline builds/service-onboarding-build-pack/Jenkinsfile]: This formatting isn't shown as screwed up when I open the file. Feel free to fix it if you can.

Comment (SauarvD) [reply builds/service-onboarding-build-pack/Jenkinsfile]: fixed formatting

File: builds/jazz_azure-create-service/components/ApiApp.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ module.exports = class ApiApp {
                 await this.resourceFactory.createHostingPlan(this.data.resourceGroupName, 'westus', {}, 'Y1', this.data.appName);
                 await this.resourceFactory.createFunctionApp(this.data.appName, storageAccountKey, this.data.tags, undefined,undefined,undefined,undefined, this.data.runtime);
                 await this.resourceFactory.uploadZipToKudu(this.data.appName, this.data.zip);
-                await this.resourceFactory.createOrUpdateApiGatewayWithSwaggerJson(this.data.serviceName, this.data.apiId, this.data.swagger, this.data.basepath);
+                await this.resourceFactory.createOrUpdateApiGatewayWithSwaggerJson(this.data.serviceName, this.data.apiId, this.data.swagger, this.data.appName, this.data.basepath);
```
Comment (raghits) [inline builds/jazz_azure-create-service/components/ApiApp.js]: Is appName a new attribute added to this method? What are the typical values? Is this only for Azure APIs?

File: builds/jazz_azure-create-service/components/ResourceFactory.js [hunk 3]
```diff
@@ -220,27 +219,55 @@ module.exports = class ResourceFactory {
...
+    }
+
     let client = await this.factory.getResource("ApiManagementClient");
+    let dataValue = await client.api.createOrUpdate(resourceGroupName, serviceName, apiId, dataParams, null);
     let result = await client.api.createOrUpdate(resourceGroupName, serviceName, apiId, parameters, null);
     this.serviceName = serviceName;
     return this.withStack(result);
   }
 
+  updateSwagger(swaggerFile, appName){
```
Comment (raghits) [inline builds/jazz_azure-create-service/components/ResourceFactory.js]: Can you add comments there to indicate what values in swagger are removed or replaced so it is easier to follow?

File: builds/jazz_azure-create-service/components/function/storageblobHandler.js [hunk 4]
```diff
@@ -1,9 +1,10 @@
 
 async function create(data, client){
   let output;
-  const blobList = await client.blobContainers.list(data.resourceGroupName, data.appName, data.resourceName);
+  const blobList = await client.blobContainers.list(data.resourceGroupName, data.appName);
```
Comment (raghits) [inline builds/jazz_azure-create-service/components/function/storageblobHandler.js]: Is resourceName no longer required?

File: builds/jazz_azure-create-service/package.json [hunk 5]
```diff
@@ -19,23 +19,24 @@
...
-    "azure-arm-storage": "^7.0.0",
-    "azure-arm-website": "^5.7.0",
+    "@azure/arm-apimanagement": "5.1.0",
+    "@azure/arm-cdn": "4.2.0",
+    "@azure/ms-rest-nodeauth": "2.0.3",
+    "@azure/arm-eventhub": "3.2.0",
+    "@azure/arm-resources": "1.1.0",
+    "@azure/arm-servicebus": "3.2.0",
+    "@azure/arm-storage": "*",
+    "azure-arm-website": "5.7.0",
```
Comment (raghits) [inline builds/jazz_azure-create-service/package.json]: Do we still need azure-arm-website?

File: builds/jazz_azure-create-service/components/ApiApp.js (hunk 2 above)
Comment (SauarvD) [reply builds/jazz_azure-create-service/components/ApiApp.js]: appName is one of the values provided in the payload. We are just passing that to this particular function in order to use it to update the swagger. Yes it is only for APIs

File: builds/jazz_azure-create-service/components/function/storageblobHandler.js (hunk 4 above)
Comment (SauarvD) [reply builds/jazz_azure-create-service/components/function/storageblobHandler.js]: after the migration to @azure/arm-* packages, resourceName is not required here as it was throwing error with this parameter

File: builds/jazz_azure-create-service/package.json (hunk 5 above)
Comment (SauarvD) [reply builds/jazz_azure-create-service/package.json]: no my bad. I will remove it.

File: builds/jazz_azure-create-service/components/ResourceFactory.js (hunk 3 above)
Comment (SauarvD) [reply builds/jazz_azure-create-service/components/ResourceFactory.js]: added description of the changes to swagger

File: core/jazz_services/components/validation/common.js [hunk 6]
```diff
@@ -351,6 +351,27 @@ var validateStatusStateChange = function (update_data, service_data_from_db, onC
     }
 };
 
+var validateProviderValue = function(service_data, onComplete) {
+    let inValidProvider = [];
+    for (let eachDeploymentAccount of service_data.deployment_accounts){
+        if(eachDeploymentAccount.provider != 'aws' && eachDeploymentAccount.provider != 'azure'){
```
Comment (raghits) [inline core/jazz_services/components/validation/common.js]: azure is invalid if the azure extension is not installed.

File: core/jazz_create-serverless-service/index.js [hunk 7]
```diff
@@ -212,6 +248,29 @@ var startServiceOnboarding = (service_creation_data, config, service_id) => {
...
+  }
+  return providerValues.every( (val, i, arr) => val === arr[0] )
+}
+
+/**
+ * Function to check and validate the list of providers
+ */
+function validateProviders(deployment_accounts){
+  for (let eachDeploymentAccount of deployment_accounts){
+    if(eachDeploymentAccount.provider != 'aws' && eachDeploymentAccount.provider != 'azure'){
```
Comment (raghits) [inline core/jazz_create-serverless-service/index.js]: `azure` is available only if azure extension is installed.

**Pull request merged**